### ISSUE TITLE
Stylelint config wizard

### ISSUE THREAD

**Issue by ai** [stylelint/stylelint#1622]
I find so awesome [CSSComb config generator](http://csscomb.com/config). And it looks not so complicated. Maybe I could find a developer to do it.

What do you think? Do we need it? Where we could put a data for it?

**Comment by ai** [comment]
Or maybe we should force users to use standard config? Like one standard is better for industry.

**Comment by jeddy3** [comment]
> Or maybe we should force users to use standard config?

This has been discussed before. We agreed that *there is no one correct config* for stylelint. If stylelint *only* had *error-checking* and *code style* rules, then perhaps there would be an argument for it. However, nearly a third of stylelint's rules are for controlling what *language features* are used on a project or by a team. What language features are allowed and disallowed is project specific and dependent on things like what browsers are being targeted, what processors are beings and what is being built e.g. a team building a webapp that only targets evergreen browsers might choose to disallow the use of the `float` property (in favour of using flexbox) e.g. `property-blacklist: ["float"]`.

This is why `stylelint-config-standard` is [designed](https://github.com/stylelint/stylelint-config-standard#stylelint-config-standard) to be a [*foundation*](https://github.com/stylelint/stylelint-config-standard#suggested-additions), of *error-checking* and *code style* rules, that teams can build upon for their particular needs. 

> I find so awesome CSSComb config generator.

This is an excellent idea and would be in-keeping with the above. I've [created an issue](https://github.com/stylelint/stylelint.io/issues/19) within the `stylelint.io` repo for it. Someone can, if they have the inclination, contribute it there.

**Comment by fleple** [comment]
I just started to do, but i still have some doubts
[my repository](https://github.com/fleple/combiner)

**Comment by ai** [comment]
@fleple I think we should promote to extend default config, instead of creating own.

**Comment by ai** [comment]
@Gaserd if you still do you own version you should show result here too

**Comment by fleple** [comment]
@ai sure

**Comment by fleple** [comment]
@ai if a understand you correctly, that the result should look like [config-standard](https://github.com/stylelint/stylelint-config-standard/blob/master/index.js), so i did it.

Or maybe you mean that the whole ```config-file``` should be initially rendered, and then it will change after each question.

Please explain me more, if im wrong and where im wrong. thank you.

**Comment by ai** [comment]
I mean it should start from `extends: "stylelint-config-standard"`

**Comment by fleple** [comment]
If you meant only this, then it's simple
```js
{
  "extends": "stylelint-config-standard",
  "rules": {
    "color-hex-case": "lower",
    "color-hex-length": "long",
    "color-named": "always-where-possible",
  },
}
```

**Comment by jeddy3** [comment]
`"color-hex-case": "lower"` [is part of](https://github.com/stylelint/stylelint-config-standard/blob/master/index.js#L23) of `stylelint-config-standard`. I think @ai is suggesting only including the rules that _override or add to_ the default settings e.g.

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "color-hex-length": "long",
    "color-named": "always-where-possible",
  },
}
```

Where:

* `"color-hex-length": "long"` _overrides_ `"color-hex-length": "short"`
* `"color-named": "always-where-possible"` is _an addition_ to the config.

**Comment by fleple** [comment]
@jeddy3 oh, thank you for explanation.

**Comment by fleple** [comment]
@ai  I think now I did according to the requirements, and now i need to add more quizzes.

**Comment by maximgatilin** [comment]
Guys, how about my version of solving this issue?
[Stylelint Config Generator](https://maximgatilin.github.io/stylelint-config/)
It's not using some cool frameworks, but it just works and have 156 stylelint rules to configure

**Comment by maximgatilin** [comment]
@jeddy3 i've [done](https://maximgatilin.github.io/stylelint-config/) this issue, how can we integrate it with stylelint site?

**Comment by ai** [comment]
@davidtheclark @jeddy3 we need your feedback about wizard and next steps :)

**Comment by jeddy3** [comment]
Thanks @maxhoffmann and @fleple for making these generators!

I'm unsure whether we should integrate them into the website though. I think they might be better suited as 3rd party tools.

I believe this will correctly signal that the stylelint org isn't responsible for keeping the generator up to date i.e. the org puts out a release (with a change of rules and/or options) and then a 3rd party config generator author can update their generator _after the fact_, rather than the stylelint org being responsible for updating the config generator _before the release_.

We took on the burden of keeping the demo update to date with each release because a. it is usually just a dependency bump b. we use the demo ourselves to quickly triage bugs. 

I suggest we add both generators to the [complementary tools lists within the dcs](https://stylelint.io/user-guide/complementary-tools/).

**Comment by maximgatilin** [comment]
I think adding link to my app in complementary tools is a good idea.

**Comment by ai** [comment]
@jeddy3 two tools can confuse users. I suggest to use @maximgatilin because of better UX.

Also I think this generator is key for better UX for new users. If we will put it in some list of tools, it will kill whole idea.

Maybe we can put it to Getting Started docs. If you want to keep distance, we can mark it as third-party tool.

But I think we should move it to the core, because it is very important tool for Stylelint promotion and new developers experience.

**Comment by ai** [comment]
Also wizard got great feedback in Twitter:

https://twitter.com/PostCSS/status/850083236155883520

> I think this is one of the most helpful engineering management tools of the year.

> Oh my gaaawd that's perfect!!!

**Comment by davidtheclark** [comment]
I agree with @ai that we should make links to this more prominent — right in the line of sight for new users.

> I think we should move it to the core

It's not going to end up in *this* repo, so the question is whether to move the tool's repo under the org or not. I'm fine either way. @jeddy3's concern is right on: I think neither he nor I can be primarily responsible for the tool — but if others on the team want to champion and maintain it, that's awesome! @maximgatilin, it's super cool that you've built this tool, and if you'd like to move it into the repo and become a @stylelint/core member, I think that would be excellent.

**Comment by ai** [comment]
@davidtheclark yeap, I mean “move it to org” instead of “move it to core [repo]” :D

**Comment by jeddy3** [comment]
>  I think neither he nor I can be primarily responsible for the tool 

I was thinking less about our maintenance burden and more about signalling. I think there's a difference between maintaining something and having something be up-to-date as a prerequisite for a release. By keeping the generator outside the org we signal that, although the tool might be useful to the user, there is a chance that it will be incompatible with the latest release. If the tool lives within the org then I think that signals that the tool _will always_ be compatible with the latest release, and if it isn't then we have user disappointment.

I think, as @ai suggested, surfacing within the Getting Started docs and labelling it as 3rd party is a good approach. It's a compromise between getting the tool in front of new users and my concerns.

**Comment by davidtheclark** [comment]
Ok, @jeddy3, let's do it. I think the signalling related to its presence outside or inside the org will not be noticed by most people, and the most important thing is that we put links in the line-of-sight for brand new users.

**Comment by sveetch** [comment]
Coming from csscomb, i was thinking to develop some tool like this just before searching in issues. Glad to see allready do it, would have be nicer to see it directly from "Complementary tools" page.

**Comment by ai** [comment]
I completely agree with @sveetch we have a great tool for new Stylelint developers, but it is hidden from users. The current website should promote it since it is a better way for new developers.

**Comment by ai** [comment]
@jeddy3 can we reconsider this issue to add the wizard to Stylelint website? 

It will really improve new developers onboarding. So it brings more users and forces them to use Stylelint better.

Maybe we can rewrite wizard to move it to the website repo?
